lib/taiko.js

```diff
@@ -3177,7 +3206,15 @@ for (const func in module.exports) {
       if (defaultConfig.observe) {
         await waitFor(defaultConfig.observeTime);
       }
-      return await realFuncs[func].apply(this, arguments);
+      try {
+        return await realFuncs[func].apply(this, arguments);
+      } catch (e) {
+        if (e.message.match(/WebSocket is not open: readyState 3/)) {
```

Comment:
`WebSocket is not open: readyState 3` may not always mean browser is crashed. It can also happen when we try to do action on closed tab if reconnect fails or after client.close() is called( could happen from taiko plugin  )

Reply:
Agreed we should combine this with some sort of check which will indicate that the `Browser` has crashed. 
> after client.close() is called( could happen from taiko plugin )

We should not allow `plugins` to use `close` method.  I.M.O we should include `close` method in [unsupported client methods list](https://github.com/getgauge/taiko/blob/3c0e34ecf36d90bdb0d329a139ae64e0d0a306e3/lib/taiko.js#L363)

Reply:
>after client.close() is called( could happen from taiko plugin ) 

while that may be true, I doubt this is what's happening in the issues reported. There are no reports of any plugin being used.

Reply:
But yes, the browser crash is only a guess, and we can be a bit defensive. The other ones are perhaps more easily reproducible.

Reply:
> But yes, the browser crash is only a guess, and we can be a bit defensive.

The first point mentioned [here](https://github.com/getgauge/taiko/pull/1115#issuecomment-595623034) will insure that Taiko gets notified whenever browser crashes.